File: plugins/jvm-plugin/src/integrationTest/resources/project/lib1/src/test/java/org/test/lib/LibOwnersTest.java
package org.test.lib;

import org.junit.Test;
import org.test.utils.LibUtils;
import org.test.utils.more.MoreUtils;

import static io.github.gmazzo.codeowners.CodeOwners.getCodeOwners;

import static kotlin.collections.SetsKt.setOf;
import static org.junit.Assert.assertEquals;

public class LibOwnersTest {

    @Test
    public void ownerOfLib() {
        assertEquals(setOf("libs-devs"), getCodeOwners(LibClass.class));
    }

    @Test
    public void ownerOfUtils() {
        assertEquals(setOf("libs-devs"), getCodeOwners(LibUtils.class));
    }

    @Test
    public void ownerOfMoreUtils() {
        assertEquals(setOf("utils-devs"), getCodeOwners(MoreUtils.class));
    }

}

File: demo-project/lib1/src/test/kotlin/org/test/lib/LibOwnersTest.kt
package org.test.lib

import io.github.gmazzo.codeowners.codeOwners
import org.test.utils.LibUtils
import org.test.utils.more.MoreUtils
import kotlin.test.Test
import kotlin.test.assertEquals

class LibOwnersTest {

    @Test
    fun ownerOfLib() {
        assertEquals(setOf("libs-devs"), LibClass::class.java.codeOwners)
    }

    @Test
    fun ownerOfUtils() {
        assertEquals(setOf("libs-devs"), LibUtils::class.java.codeOwners)
    }

    @Test
    fun ownerOfMoreUtils() {
        assertEquals(setOf("utils-devs"), MoreUtils::class.java.codeOwners)
        assertEquals(setOf("utils-more"), MoreUtils.Companion::class.java.codeOwners)
    }

}
